config/config.go

```diff
@@ -180,38 +181,73 @@ type Config struct {
 	OmitExtensions    bool
 }
 
+func extractMasterKeys(group keyGroup) (sops.KeyGroup, error) {
+	var mergeKeyGroup sops.KeyGroup
+	for _, k := range group.Merge {
+		subKeyGroup, err := extractMasterKeys(k)
+		if err != nil {
+			return nil, err
+		}
+		mergeKeyGroup = append(mergeKeyGroup, subKeyGroup...)
+	}
+
+	var keyGroup sops.KeyGroup
+
+	// Add only unique values from mergeGroup to keyGroup
+	unique := make(map[string]bool)
+	for _, v := range mergeKeyGroup {
+		// keytype + toMap make a unique combination per key type
+		valueMap := v.ToMap()
+		delete(valueMap, "created_at")
+
+		key := fmt.Sprintf("%T/%v", v, valueMap)
+		if _, ok := unique[key]; ok {
+			// key already contained, therefore not unique
+			continue
+		}
+
+		keyGroup = append(keyGroup, v)
+		unique[key] = true
+	}
```

Comment:
How about moving the deduplication code to a function
```go
func deduplicateKeygroup(group keyGroup) sops.KeyGroup {
  ...
}
```
? That would make `extractMasterKeys` easier to read, and the deduplication code easier to move around.